### PR TITLE
Convert okhttp-sse to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp-sse/build.gradle [hunk 1]
```diff
@@ -16,3 +16,18 @@ dependencies {
   testImplementation deps.assertj
   testCompileOnly deps.jsr305
 }
+
+apply plugin: 'me.champeau.gradle.japicmp'
```
Comment (Egorand) [inline okhttp-sse/build.gradle]: Move to line 2

File: okhttp-sse/src/main/java/okhttp3/sse/EventSources.kt [hunk 2]
```diff
@@ -13,27 +13,28 @@
...
-      return eventSource;
-    };
+object EventSources {
+  @JvmStatic
+  fun createFactory(client: OkHttpClient): EventSource.Factory {
+    return object : EventSource.Factory {
+      override fun newEventSource(request: Request, listener: EventSourceListener): EventSource {
+        val eventSource = RealEventSource(request, listener)
+        eventSource.connect(client)
+        return eventSource
```
Comment (Egorand) [inline okhttp-sse/src/main/java/okhttp3/sse/EventSources.kt]: nit: `return RealEventSource(request, listener).apply { connect(client) }`

File: okhttp-sse/build.gradle (hunk 1 above)
Comment (swankjesse) [reply okhttp-sse/build.gradle]: Fix coming

File: okhttp-sse/src/main/java/okhttp3/sse/EventSources.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp-sse/src/main/java/okhttp3/sse/EventSources.kt]: Fix coming!

File: okhttp-sse/build.gradle (hunk 1 above)
Comment (swankjesse) [reply okhttp-sse/build.gradle]: https://github.com/square/okhttp/pull/4898

File: okhttp-sse/src/main/java/okhttp3/sse/EventSources.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp-sse/src/main/java/okhttp3/sse/EventSources.kt]: https://github.com/square/okhttp/pull/4899

**Pull request merged**